exercises/run-length-encoding/tests/run-length-encoding.rs

```diff
@@ -0,0 +1,87 @@
+extern crate run_length_encoding as cipher;
+
+// encoding tests
+
+#[test]
+// #[ignore]
+fn test_encode_empty_string() {
+    assert_eq!("", cipher::encode(""));
+}
+
+#[test]
+// #[ignore]
+fn test_encode_single_characters() {
+    assert_eq!("XYZ", cipher::encode("XYZ"));
+}
+
+#[test]
+// #[ignore]
+fn test_encode_string_with_no_single_characters() {
+    assert_eq!("2A3B4C", cipher::encode("AABBBCCCC"));
+}
+
+#[test]
+// #[ignore]
+fn test_encode_single_characters_mixed_with_repeated_characters() {
+    assert_eq!("12WB12W3B24WB",cipher::encode(
+        "WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB"));
+}
+
+#[test]
+// #[ignore]
+fn test_encode_multiple_whitespace_mixed_in_string() {
+    assert_eq!("2 hs2q q2w2 ", cipher::encode("  hsqq qww  "));
+}
+
+#[test]
+// #[ignore]
+fn test_encode_lowercase_characters() {
+    assert_eq!("2a3b4c", cipher::encode("aabbbcccc"));
+}
+
+// decoding tests
+
+#[test]
+// #[ignore]
+fn test_decode_empty_string() {
+    assert_eq!("" , cipher::decode(""));
+}
+
+#[test]
+// #[ignore]
+fn test_decode_single_characters_only() {
+    assert_eq!("XYZ", cipher::decode("XYZ"));
+}
+
+#[test]
+// #[ignore]
+fn test_decode_string_with_no_single_characters() {
+    assert_eq!("AABBBCCCC" , cipher::decode("2A3B4C"));
```

Comment:
there's a space before the comma here, but not in most other cases. Should it be removed?